server/testdriver/core/src/main/java/org/infinispan/server/test/core/persistence/ContainerDatabase.java

```diff
@@ -168,6 +169,14 @@ public String getDriverClassName() {
          return database.driverClassName();
       }
 
+      @Override
+      protected void waitUntilContainerStarted() {
+         // Override JdbcContainerAdapter#waitUntulContainerStarted to ensure that any additional waitStrategies defined,
+         // e.g. LogMessageWaitStrategy, are respected
+         getWaitStrategy().waitUntilReady(this);
```

Comment:
This probably needs to be conditional and only if `getWaitStrategy() != null`. We are setting waitStrategy only if `org.infinispan.test.database.container.log.regex` specified

Reply:
We should be good as `getWaitStrategy` never returns null as the implementation is:

```java
   protected WaitStrategy getWaitStrategy() {
        return this.waitStrategy == DEFAULT_WAIT_STRATEGY ? this.containerDef.getWaitStrategy() : this.waitStrategy;
    }
```

and `containerDef` is always initialized to `GenericContainer.DEFAULT_WAIT_STRATEGY`